doc/compiler/aot/InlinedMethods.md

```diff
@@ -0,0 +1,259 @@
+the list is associated with the first inlined site.
+
+The reason for ensuring this is because validating/materialize values for
+subsequent inlined methods may depend on the previous entry. For example,
+if `foo() calls bar() calls baz()`, and `bar` and `baz` are inlined (first
+`bar` would be inlined, then `baz`), `baz` is named in `bar`'s constant
+pool, and `bar` is named in `foo`'s constant pool. Therefore, in order to
+materialize the `J9Method` of `baz`, one first needs the `J9Method` of `bar`,
+which first requires the `J9Method` of `foo`.
+
+Generating external relocations for inlined methods requires the 
+existence of Guards. Guards are used to gate execution of regions of 
+code. When a method is inlined, a guard is generated for that inlined 
+method. It is possible for a guard to be removed for various reasons. 
+Currently, an inlined method cannot be validated/relocated without an 
+associated guard<sup>1</sup>. As such, if a guard is removed, it is 
+still kept aside so that the information it holds can be used 
+to create an external relocation. If a guard is removed, the Inlined Method 
+/ Profiled Inlined Method relocations are generated; if the guard remains, 
+the Inlined Method with NOP Guard / Profiled Inlined Method with Guard 
+relocations are generated.
+
+Unless the [SVM](https://github.com/eclipse/openj9/blob/master/doc/compiler/aot/SymbolValidationManager.md)
+is enabled, the inlined method external relocations are the
+last entries added to the list; if the SVM is enabled, the SVM validation
+records are added next - part of the validation that would've been done by
+the inlined method validation procedure is delegated to the SVM.
+
+# Validating Inlined Sites 
+
+The way an inlined method is validated depends on whether it was
+profiled or not.
+
+## Inlined Methods with or without NOP Guards
+
+The process starts in `TR_RelocationRecordInlinedMethod::preparePrivateData`.
+First `TR_RelocationRecordInlinedMethod::inlinedSiteValid` is called, which
+
+1. Gets the J9Method of the caller. For the very first inlined site, this will be the method being compiled; for the rest it will be from one of the previously validated/relocated inlined sites.
+2. Checks if the method has been unloaded.
+3. Materializes the J9Method of the inlined method:
+    * If the SVM is enabled the method ID in the relocation record is used.
+    * If the SVM is disabled, `getMethodFromCP` is called which is a virtual method that has different implementations depending on the kind of inlined method (special, static, virtual, interface).
+4. Checks if the inlined site can be activated because of debug or other JVM restrictions.
+5. Checks if J9ROMClass of the J9Method from step 3 matches the J9ROMClass in the SCC based on the offset stored in the relocation record.
+
+If everything succeeds, `inlinedSiteValid` returns true. The success or
+failure is cached in `reloPrivateData->_failValidation`.
```

Comment:
Side note we should think about renaming `_failValidation` to `validationResult` if both success and failure can be recorded.